README.md
# AFT survival regression using Rate Mixtures of Weibull distributions

The `RMWreg` library is provided as supplementary material to [Vallejos and Steel (2017)](http://www.sciencedirect.com/science/article/pii/S2452306217300072). 

# Funding acknowledgment 

We acknowledge support from the University of Warwick, the Pontificia Universidad Católica de Chile and the Alan Turing Institute (EPRSC grant no EP/N510129/1). 






NEWS
version 0.0.2: (2016-08-25)
- Full MCMC and reduced variations running for Weibul model

version 0.0.3: (2016-08-26)
- Template for generic RMWreg MCMC sampler has been added

version 0.0.4: (2016-09-15)
- RMWreg_MCMC function added (with advance options allowed)
- Updates of mixing parameters are not yet implemented

version 0.0.5: (2016-09-16)
- RMWreg_MCMC with Mixing = "None" is working
- RMWreg_MCMC with Mixing = "Exponential" is working

version 0.0.6: (2016-09-20)
- RMWreg_MCMC with Mixing = "Gamma" is working
- RMWreg_MCMC with Mixing = "InvGamma" is working
- RMWreg_MCMC with Mixing = "InvGauss" is working

version 0.0.7: (2016-09-23)
- RMWreg_MCMC with Mixing = "LogNormal" is working
- Function RMWreg_DIC has been added
- Function RMWreg_CaseDeletion has been added
- Function RMWreg_logML has been added

version 0.0.8: (2016-09-25)
- Loops replaces by apply in RMWreg_DIC and RMWreg_CaseDeletion for Mixing = "LogNormal"
- Function RMWreg_BFoutlier has been added (Exponential mixing only for now)

version 0.0.9: (2016-09-25)
- Small changes to RMWreg_logML function

version 0.0.10: (2016-09-29)
- Fixed bug in RMWreg_logML function

version 0.0.11: (2016-09-29)
- Fixed bug in RMWreg_logML function

version 0.0.12: (2016-09-30)
- Fixed bug in RMWreg_logML function (for Mixing = "InvGamma")

version 0.0.13: (2016-10-05)
- Fixed bug in 'cvStarSquare' function (c++ code, only affects log-normal mixing)

version 0.0.14: (2016-10-06)
- Fixed issues in MCMC sampler for log-normal mixing (change of parametrization)

version 0.0.15: (2016-10-06)
- Fixed issues in model comparison for log-normal mixing (change of parametrization)

version 0.0.16: (2016-10-07)
- Storage of 'lambda' chain added to 'RMWreg_MCMC' function

version 0.0.17: (2016-10-10)
- Fixed bug in 'RMWreg_BFoutlier' function

version 0.0.18: (2016-10-13)
- Small changes to 'RMWreg_logML' function

version 0.0.19: (2017-03-09)
- Template added for 'RMWreg_PredictCDF' function.

version 0.0.20: (2017-03-09)
- 'RMWreg_PredictCDF' function working for BaseModel = 'Weibull' and Mixing = 'None'

version 0.0.21: (2017-03-19)
- Update in 'RMWreg_MCMC' function to allow user-defined starting values

version 0.0.22: (2019-03-05)
- Small updates in DESCRIPTION (R version requirement and needs compilation)

TO-DO
- Extend RMWreg_BFoutlier to the remaining mixing distributions



R/RcppExports.R
# Generated by using Rcpp::compileAttributes() -> do not edit by hand
# Generator token: 10BE3573-1514-4C36-9D1C-5A225CD40393

HiddenLogPriorTheta <- function(theta, gam, a, type, mixing) {
    .Call('_RMWreg_HiddenLogPriorTheta', PACKAGE = 'RMWreg', theta, gam, a, type, mixing)
}

HiddenRMWreg_MCMC <- function(N, thin, burn, Time, Event, X, mixing, Hyp1Gam, Hyp2Gam, PriorCV, HypTheta, beta0, gam0, theta0, Adapt, ar, StoreAdapt, EndAdapt, LSbeta0, LSgam0, LStheta0, FixBetaJ, FixGam, FixTheta, PrintProgress, lambdaPeriod, FixLambdaI, RefLambda) {
    .Call('_RMWreg_HiddenRMWreg_MCMC', PACKAGE = 'RMWreg', N, thin, burn, Time, Event, X, mixing, Hyp1Gam, Hyp2Gam, PriorCV, HypTheta, beta0, gam0, theta0, Adapt, ar, StoreAdapt, EndAdapt, LSbeta0, LSgam0, LStheta0, FixBetaJ, FixGam, FixTheta, PrintProgress, lambdaPeriod, FixLambdaI, RefLambda)
}

HiddenLogLik <- function(Time, Event, DesignMat, beta, gam, theta, Mixing, BaseModel) {
    .Call('_RMWreg_HiddenLogLik', PACKAGE = 'RMWreg', Time, Event, DesignMat, beta, gam, theta, Mixing, BaseModel)
}

HiddenRMWreg_DIC <- function(Chain, Time, Event, DesignMat, Mixing, BaseModel) {
    .Call('_RMWreg_HiddenRMWreg_DIC', PACKAGE = 'RMWreg', Chain, Time, Event, DesignMat, Mixing, BaseModel)
}

HiddenRMWreg_CaseDeletion <- function(Chain, Time, Event, DesignMat, Mixing, BaseModel) {
    .Call('_RMWreg_HiddenRMWreg_CaseDeletion', PACKAGE = 'RMWreg', Chain, Time, Event, DesignMat, Mixing, BaseModel)
}

HiddenRMWreg_BFoutlierObs <- function(Chain, Obs, RefLambda, Time, Event, DesignMat, PriorCV, HypTheta, Hyp1Gam, Hyp2Gam, Mixing, BaseModel, thin, lambdaPeriod, ar) {
    .Call('_RMWreg_HiddenRMWreg_BFoutlierObs', PACKAGE = 'RMWreg', Chain, Obs, RefLambda, Time, Event, DesignMat, PriorCV, HypTheta, Hyp1Gam, Hyp2Gam, Mixing, BaseModel, thin, lambdaPeriod, ar)
}

HiddenRMWreg_BFoutlier <- function(Chain, RefLambda, Time, Event, DesignMat, PriorCV, HypTheta, Hyp1Gam, Hyp2Gam, Mixing, BaseModel, thin, lambdaPeriod, ar) {
    .Call('_RMWreg_HiddenRMWreg_BFoutlier', PACKAGE = 'RMWreg', Chain, RefLambda, Time, Event, DesignMat, PriorCV, HypTheta, Hyp1Gam, Hyp2Gam, Mixing, BaseModel, thin, lambdaPeriod, ar)
}



R/User_Codes_aux_Delete.R
###############################################################################
###############################################################################
######################## ALL CODES FOR Weibull  MODEL #########################
###############################################################################
###############################################################################


###############################################################################
###############################################################################
######################## ALL CODES FOR RMWEXP   MODEL #########################
###############################################################################
###############################################################################



###############################################################################
###############################################################################
######################## ALL CODES FOR RMWGAM   MODEL #########################
###############################################################################
###############################################################################

################################################################################
# OUTLIER DETECTION
BF.lambda.obs.RMWGAM<-function(ref,obs,N,thin,Q,burn,Time,Cens,X,chain,typ.theta,hyp.theta,hyp1.gam=1,hyp2.gam=1,ar=0.44,EXP=FALSE)
{
  #######################################################################
  # IMPORTANT: THIS TAKES AS ARGUMENT THE CHAIN AFTER BURN.
  #######################################################################
  chain=as.matrix(chain)
  aux1=Post.lambda.obs.RMWGAM(ref,obs,Time,Cens,X,chain)
  aux2=CFP.obs.RMWGAM(ref,obs,N,thin,Q,burn,Time,Cens,X,chain,typ.theta,hyp.theta,hyp1.gam,hyp2.gam,ar,EXP)
  return(aux1*aux2)
}


###############################################################################
###############################################################################
######################## ALL CODES FOR RMWIGAM  MODEL #########################
###############################################################################
###############################################################################

################################################################################
# CORRECTION FACTOR FOR OUTLIER DETECTION RMWIGAM MODELS
OD.Correction.RMWIGAM<-function(chain,Time,X)
{
  #######################################################################
  # IMPORTANT: THIS TAKES AS ARGUMENT THE CHAIN AFTER BURN.
  #######################################################################
  chain=as.matrix(chain)
  k=dim(X)[2]
  if(k>1)  {beta=apply(chain[,1:k],2,"median")}
  else		{beta=median(chain[,1])}
  gam=median(chain[,k+1]); theta=median(chain[,k+2])
  aux0=2*sqrt((exp(-X%*%beta)*Time)^gam)
  aux=(besselK(x=aux0,nu=-theta+1,expon.scaled = FALSE)^2)/(besselK(x=aux0,nu=-theta+2,expon.scaled = FALSE)*besselK(x=aux0,nu=-theta,expon.scaled = FALSE))
  return(aux)
}

################################################################################
# OUTLIER DETECTION
BF.lambda.obs.RMWIGAM<-function(ref,obs,N,thin,Q,burn,Time,Cens,X,chain,typ.theta,hyp.theta,hyp1.gam=1,hyp2.gam=1,ar=0.44,EXP=FALSE)
{
  #######################################################################
  # IMPORTANT: THIS TAKES AS ARGUMENT THE CHAIN AFTER BURN.
  #######################################################################
  chain=as.matrix(chain)
  aux1=Post.lambda.obs.RMWIGAM(ref,obs,Time,Cens,X,chain)
  aux2=CFP.obs.RMWIGAM(ref,obs,N,thin,Q,burn,Time,Cens,X,chain,typ.theta,hyp.theta,hyp1.gam,hyp2.gam,ar,EXP)
  return(aux1*aux2)
}


###############################################################################
###############################################################################
######################## ALL CODES FOR RMWIGAUSSMODEL #########################
###############################################################################
###############################################################################

################################################################################
# CORRECTION FACTOR FOR OUTLIER DETECTION RMWIGAUSS MODELS
OD.Correction.RMWIGAUSS<-function(chain,Time,X)
{
  #######################################################################
  # IMPORTANT: THIS TAKES AS ARGUMENT THE CHAIN AFTER BURN.
  #######################################################################
  chain=as.matrix(chain)
  k=dim(X)[2]
  if(k>1)  {beta=apply(chain[,1:k],2,"median")}
  else  	{beta=median(chain[,1])}
  gam=median(chain[,k+1]); theta=median(chain[,k+2])
  aux0=sqrt(2*(exp(-X%*%beta)*Time)^gam+theta^(-2))
  aux=(besselK(x=aux0,nu=1/2,expon.scaled = FALSE)^2)/(besselK(x=aux0,nu=-1/2,expon.scaled = FALSE)*besselK(x=aux0,nu=3/2,expon.scaled = FALSE))
  return(aux)
}

################################################################################
# OUTLIER DETECTION
BF.lambda.obs.RMWIGAUSS<-function(ref,obs,N,thin,Q,burn,Time,Cens,X,chain,typ.theta,hyp.theta,hyp1.gam=1,hyp2.gam=1,ar=0.44,EXP=FALSE)
{
  #######################################################################
  # IMPORTANT: THIS TAKES AS ARGUMENT THE CHAIN AFTER BURN.
  #######################################################################
  chain=as.matrix(chain)
  aux1=Post.lambda.obs.RMWIGAUSS(ref,obs,Time,Cens,X,chain)
  aux2=CFP.obs.RMWIGAUSS(ref,obs,N,thin,Q,burn,Time,Cens,X,chain,typ.theta,hyp.theta,hyp1.gam,hyp2.gam,ar,EXP)
  return(aux1*aux2)
}

###############################################################################
###############################################################################
######################## ALL CODES FOR RMWLN    MODEL #########################
###############################################################################
###############################################################################

################################################################################
# CORRECTION FACTOR FOR OUTLIER DETECTION RMWLN MODEL
OD.Correction.RMWLN<-function(chain,Time,X)
{
  #######################################################################
  # IMPORTANT: THIS TAKES AS ARGUMENT THE CHAIN AFTER BURN.
  #######################################################################
  chain=as.matrix(chain)
  n=length(Time); k=dim(X)[2]
  if(k>1)	{beta=apply(chain[,1:k],2,"median")}
  else		{beta=median(chain[,1])}
  gam=median(chain[,k+1]); theta=median(chain[,k+2])
  R0=(exp(-X%*%beta)*Time)^gam
  norm0=rep(0,times=n); norm1=rep(0,times=n)
  exp0=rep(0,times=n); exp1=rep(0,times=n)
  for(i in 1:n)
  {
    norm0[i]=integrate(f.lambda.aux,lower=0,upper=Inf,RATE=R0[i],theta=theta,Cens=0)$value
    norm1[i]=integrate(f.lambda.aux,lower=0,upper=Inf,RATE=R0[i],theta=theta,Cens=1)$value
    exp0[i]=integrate(E.lambda.aux,lower=0,upper=Inf,RATE=R0[i],theta=theta,Cens=0)$value
    exp1[i]=integrate(E.lambda.aux,lower=0,upper=Inf,RATE=R0[i],theta=theta,Cens=1)$value
  }
  aux=(norm1/norm0)*(exp0/exp1)
  return(aux)
}

################################################################################
# OUTLIER DETECTION
BF.lambda.obs.RMWLN<-function(ref,obs,N,thin,Q,burn,Time,Cens,X,chain,typ.theta,hyp.theta,hyp1.gam=1,hyp2.gam=1,ar=0.44,EXP=FALSE)
{
  #######################################################################
  # IMPORTANT: THIS TAKES AS ARGUMENT THE CHAIN AFTER BURN.
  #######################################################################
  chain=as.matrix(chain)
  aux1=Post.lambda.obs.RMWLN(ref,obs,Time,Cens,X,chain)
  aux2=CFP.obs.RMWLN(ref,obs,N,thin,Q,burn,Time,Cens,X,chain,typ.theta,hyp.theta,hyp1.gam,hyp2.gam,ar,EXP)
  return(aux1*aux2)
}


src/RcppExports.cpp
// Generated by using Rcpp::compileAttributes() -> do not edit by hand
// Generator token: 10BE3573-1514-4C36-9D1C-5A225CD40393

#include <RcppArmadillo.h>
#include <Rcpp.h>

using namespace Rcpp;

// HiddenLogPriorTheta
double HiddenLogPriorTheta(double const& theta, double const& gam, double const& a, String const& type, String const& mixing);
RcppExport SEXP _RMWreg_HiddenLogPriorTheta(SEXP thetaSEXP, SEXP gamSEXP, SEXP aSEXP, SEXP typeSEXP, SEXP mixingSEXP) {
BEGIN_RCPP
    Rcpp::RObject rcpp_result_gen;
    Rcpp::RNGScope rcpp_rngScope_gen;
    Rcpp::traits::input_parameter< double const& >::type theta(thetaSEXP);
    Rcpp::traits::input_parameter< double const& >::type gam(gamSEXP);
    Rcpp::traits::input_parameter< double const& >::type a(aSEXP);
    Rcpp::traits::input_parameter< String const& >::type type(typeSEXP);
    Rcpp::traits::input_parameter< String const& >::type mixing(mixingSEXP);
    rcpp_result_gen = Rcpp::wrap(HiddenLogPriorTheta(theta, gam, a, type, mixing));
    return rcpp_result_gen;
END_RCPP
}
// HiddenRMWreg_MCMC
Rcpp::List HiddenRMWreg_MCMC(int N, int thin, int burn, NumericVector Time, NumericVector Event, NumericMatrix X, String mixing, double const& Hyp1Gam, double const& Hyp2Gam, String const& PriorCV, double const& HypTheta, NumericVector beta0, double gam0, double theta0, int Adapt, double ar, int StoreAdapt, int EndAdapt, NumericVector LSbeta0, double LSgam0, double LStheta0, int FixBetaJ, int FixGam, int FixTheta, int PrintProgress, int lambdaPeriod, int FixLambdaI, double RefLambda);
RcppExport SEXP _RMWreg_HiddenRMWreg_MCMC(SEXP NSEXP, SEXP thinSEXP, SEXP burnSEXP, SEXP TimeSEXP, SEXP EventSEXP, SEXP XSEXP, SEXP mixingSEXP, SEXP Hyp1GamSEXP, SEXP Hyp2GamSEXP, SEXP PriorCVSEXP, SEXP HypThetaSEXP, SEXP beta0SEXP, SEXP gam0SEXP, SEXP theta0SEXP, SEXP AdaptSEXP, SEXP arSEXP, SEXP StoreAdaptSEXP, SEXP EndAdaptSEXP, SEXP LSbeta0SEXP, SEXP LSgam0SEXP, SEXP LStheta0SEXP, SEXP FixBetaJSEXP, SEXP FixGamSEXP, SEXP FixThetaSEXP, SEXP PrintProgressSEXP, SEXP lambdaPeriodSEXP, SEXP FixLambdaISEXP, SEXP RefLambdaSEXP) {
BEGIN_RCPP
    Rcpp::RObject rcpp_result_gen;
    Rcpp::RNGScope rcpp_rngScope_gen;
    Rcpp::traits::input_parameter< int >::type N(NSEXP);
    Rcpp::traits::input_parameter< int >::type thin(thinSEXP);
    Rcpp::traits::input_parameter< int >::type burn(burnSEXP);
    Rcpp::traits::input_parameter< NumericVector >::type Time(TimeSEXP);
    Rcpp::traits::input_parameter< NumericVector >::type Event(EventSEXP);
    Rcpp::traits::input_parameter< NumericMatrix >::type X(XSEXP);
    Rcpp::traits::input_parameter< String >::type mixing(mixingSEXP);
    Rcpp::traits::input_parameter< double const& >::type Hyp1Gam(Hyp1GamSEXP);
    Rcpp::traits::input_parameter< double const& >::type Hyp2Gam(Hyp2GamSEXP);
    Rcpp::traits::input_parameter< String const& >::type PriorCV(PriorCVSEXP);
    Rcpp::traits::input_parameter< double const& >::type HypTheta(HypThetaSEXP);
    Rcpp::traits::input_parameter< NumericVector >::type beta0(beta0SEXP);
    Rcpp::traits::input_parameter< double >::type gam0(gam0SEXP);
    Rcpp::traits::input_parameter< double >::type theta0(theta0SEXP);
    Rcpp::traits::input_parameter< int >::type Adapt(AdaptSEXP);
    Rcpp::traits::input_parameter< double >::type ar(arSEXP);
    Rcpp::traits::input_parameter< int >::type StoreAdapt(StoreAdaptSEXP);
    Rcpp::traits::input_parameter< int >::type EndAdapt(EndAdaptSEXP);
    Rcpp::traits::input_parameter< NumericVector >::type LSbeta0(LSbeta0SEXP);
    Rcpp::traits::input_parameter< double >::type LSgam0(LSgam0SEXP);
    Rcpp::traits::input_parameter< double >::type LStheta0(LStheta0SEXP);
    Rcpp::traits::input_parameter< int >::type FixBetaJ(FixBetaJSEXP);
    Rcpp::traits::input_parameter< int >::type FixGam(FixGamSEXP);
    Rcpp::traits::input_parameter< int >::type FixTheta(FixThetaSEXP);
    Rcpp::traits::input_parameter< int >::type PrintProgress(PrintProgressSEXP);
    Rcpp::traits::input_parameter< int >::type lambdaPeriod(lambdaPeriodSEXP);
    Rcpp::traits::input_parameter< int >::type FixLambdaI(FixLambdaISEXP);
    Rcpp::traits::input_parameter< double >::type RefLambda(RefLambdaSEXP);
    rcpp_result_gen = Rcpp::wrap(HiddenRMWreg_MCMC(N, thin, burn, Time, Event, X, mixing, Hyp1Gam, Hyp2Gam, PriorCV, HypTheta, beta0, gam0, theta0, Adapt, ar, StoreAdapt, EndAdapt, LSbeta0, LSgam0, LStheta0, FixBetaJ, FixGam, FixTheta, PrintProgress, lambdaPeriod, FixLambdaI, RefLambda));
    return rcpp_result_gen;
END_RCPP
}
// HiddenLogLik
double HiddenLogLik(arma::vec const& Time, arma::vec const& Event, arma::mat const& DesignMat, arma::vec const& beta, double const& gam, double const& theta, String const& Mixing, String const& BaseModel);
RcppExport SEXP _RMWreg_HiddenLogLik(SEXP TimeSEXP, SEXP EventSEXP, SEXP DesignMatSEXP, SEXP betaSEXP, SEXP gamSEXP, SEXP thetaSEXP, SEXP MixingSEXP, SEXP BaseModelSEXP) {
BEGIN_RCPP
    Rcpp::RObject rcpp_result_gen;
    Rcpp::RNGScope rcpp_rngScope_gen;
    Rcpp::traits::input_parameter< arma::vec const& >::type Time(TimeSEXP);
    Rcpp::traits::input_parameter< arma::vec const& >::type Event(EventSEXP);
    Rcpp::traits::input_parameter< arma::mat const& >::type DesignMat(DesignMatSEXP);
    Rcpp::traits::input_parameter< arma::vec const& >::type beta(betaSEXP);
    Rcpp::traits::input_parameter< double const& >::type gam(gamSEXP);
    Rcpp::traits::input_parameter< double const& >::type theta(thetaSEXP);
    Rcpp::traits::input_parameter< String const& >::type Mixing(MixingSEXP);
    Rcpp::traits::input_parameter< String const& >::type BaseModel(BaseModelSEXP);
    rcpp_result_gen = Rcpp::wrap(HiddenLogLik(Time, Event, DesignMat, beta, gam, theta, Mixing, BaseModel));
    return rcpp_result_gen;
END_RCPP
}
// HiddenRMWreg_DIC
double HiddenRMWreg_DIC(Rcpp::List const& Chain, arma::vec const& Time, arma::vec const& Event, arma::mat const& DesignMat, String const& Mixing, String const& BaseModel);
RcppExport SEXP _RMWreg_HiddenRMWreg_DIC(SEXP ChainSEXP, SEXP TimeSEXP, SEXP EventSEXP, SEXP DesignMatSEXP, SEXP MixingSEXP, SEXP BaseModelSEXP) {
BEGIN_RCPP
    Rcpp::RObject rcpp_result_gen;
    Rcpp::RNGScope rcpp_rngScope_gen;
    Rcpp::traits::input_parameter< Rcpp::List const& >::type Chain(ChainSEXP);
    Rcpp::traits::input_parameter< arma::vec const& >::type Time(TimeSEXP);
    Rcpp::traits::input_parameter< arma::vec const& >::type Event(EventSEXP);
    Rcpp::traits::input_parameter< arma::mat const& >::type DesignMat(DesignMatSEXP);
    Rcpp::traits::input_parameter< String const& >::type Mixing(MixingSEXP);
    Rcpp::traits::input_parameter< String const& >::type BaseModel(BaseModelSEXP);
    rcpp_result_gen = Rcpp::wrap(HiddenRMWreg_DIC(Chain, Time, Event, DesignMat, Mixing, BaseModel));
    return rcpp_result_gen;
END_RCPP
}
// HiddenRMWreg_CaseDeletion
arma::mat HiddenRMWreg_CaseDeletion(Rcpp::List const& Chain, arma::vec const& Time, arma::vec const& Event, arma::mat const& DesignMat, String const& Mixing, String const& BaseModel);
RcppExport SEXP _RMWreg_HiddenRMWreg_CaseDeletion(SEXP ChainSEXP, SEXP TimeSEXP, SEXP EventSEXP, SEXP DesignMatSEXP, SEXP MixingSEXP, SEXP BaseModelSEXP) {
BEGIN_RCPP
    Rcpp::RObject rcpp_result_gen;
    Rcpp::RNGScope rcpp_rngScope_gen;
    Rcpp::traits::input_parameter< Rcpp::List const& >::type Chain(ChainSEXP);
    Rcpp::traits::input_parameter< arma::vec const& >::type Time(TimeSEXP);
    Rcpp::traits::input_parameter< arma::vec const& >::type Event(EventSEXP);
    Rcpp::traits::input_parameter< arma::mat const& >::type DesignMat(DesignMatSEXP);
    Rcpp::traits::input_parameter< String const& >::type Mixing(MixingSEXP);
    Rcpp::traits::input_parameter< String const& >::type BaseModel(BaseModelSEXP);
    rcpp_result_gen = Rcpp::wrap(HiddenRMWreg_CaseDeletion(Chain, Time, Event, DesignMat, Mixing, BaseModel));
    return rcpp_result_gen;
END_RCPP
}
// HiddenRMWreg_BFoutlierObs
double HiddenRMWreg_BFoutlierObs(Rcpp::List Chain, int const& Obs, arma::vec const& RefLambda, arma::vec const& Time, arma::vec const& Event, arma::mat const& DesignMat, String const& PriorCV, double const& HypTheta, double const& Hyp1Gam, double const& Hyp2Gam, String const& Mixing, String const& BaseModel, int thin, int lambdaPeriod, double ar);
RcppExport SEXP _RMWreg_HiddenRMWreg_BFoutlierObs(SEXP ChainSEXP, SEXP ObsSEXP, SEXP RefLambdaSEXP, SEXP TimeSEXP, SEXP EventSEXP, SEXP DesignMatSEXP, SEXP PriorCVSEXP, SEXP HypThetaSEXP, SEXP Hyp1GamSEXP, SEXP Hyp2GamSEXP, SEXP MixingSEXP, SEXP BaseModelSEXP, SEXP thinSEXP, SEXP lambdaPeriodSEXP, SEXP arSEXP) {
BEGIN_RCPP
    Rcpp::RObject rcpp_result_gen;
    Rcpp::RNGScope rcpp_rngScope_gen;
    Rcpp::traits::input_parameter< Rcpp::List >::type Chain(ChainSEXP);
    Rcpp::traits::input_parameter< int const& >::type Obs(ObsSEXP);
    Rcpp::traits::input_parameter< arma::vec const& >::type RefLambda(RefLambdaSEXP);
    Rcpp::traits::input_parameter< arma::vec const& >::type Time(TimeSEXP);
    Rcpp::traits::input_parameter< arma::vec const& >::type Event(EventSEXP);
    Rcpp::traits::input_parameter< arma::mat const& >::type DesignMat(DesignMatSEXP);
    Rcpp::traits::input_parameter< String const& >::type PriorCV(PriorCVSEXP);
    Rcpp::traits::input_parameter< double const& >::type HypTheta(HypThetaSEXP);
    Rcpp::traits::input_parameter< double const& >::type Hyp1Gam(Hyp1GamSEXP);
    Rcpp::traits::input_parameter< double const& >::type Hyp2Gam(Hyp2GamSEXP);
    Rcpp::traits::input_parameter< String const& >::type Mixing(MixingSEXP);
    Rcpp::traits::input_parameter< String const& >::type BaseModel(BaseModelSEXP);
    Rcpp::traits::input_parameter< int >::type thin(thinSEXP);
    Rcpp::traits::input_parameter< int >::type lambdaPeriod(lambdaPeriodSEXP);
    Rcpp::traits::input_parameter< double >::type ar(arSEXP);
    rcpp_result_gen = Rcpp::wrap(HiddenRMWreg_BFoutlierObs(Chain, Obs, RefLambda, Time, Event, DesignMat, PriorCV, HypTheta, Hyp1Gam, Hyp2Gam, Mixing, BaseModel, thin, lambdaPeriod, ar));
    return rcpp_result_gen;
END_RCPP
}
// HiddenRMWreg_BFoutlier
arma::vec HiddenRMWreg_BFoutlier(Rcpp::List Chain, arma::vec const& RefLambda, arma::vec const& Time, arma::vec const& Event, arma::mat const& DesignMat, String const& PriorCV, double const& HypTheta, double const& Hyp1Gam, double const& Hyp2Gam, String const& Mixing, String const& BaseModel, int thin, int lambdaPeriod, double ar);
RcppExport SEXP _RMWreg_HiddenRMWreg_BFoutlier(SEXP ChainSEXP, SEXP RefLambdaSEXP, SEXP TimeSEXP, SEXP EventSEXP, SEXP DesignMatSEXP, SEXP PriorCVSEXP, SEXP HypThetaSEXP, SEXP Hyp1GamSEXP, SEXP Hyp2GamSEXP, SEXP MixingSEXP, SEXP BaseModelSEXP, SEXP thinSEXP, SEXP lambdaPeriodSEXP, SEXP arSEXP) {
BEGIN_RCPP
    Rcpp::RObject rcpp_result_gen;
    Rcpp::RNGScope rcpp_rngScope_gen;
    Rcpp::traits::input_parameter< Rcpp::List >::type Chain(ChainSEXP);
    Rcpp::traits::input_parameter< arma::vec const& >::type RefLambda(RefLambdaSEXP);
    Rcpp::traits::input_parameter< arma::vec const& >::type Time(TimeSEXP);
    Rcpp::traits::input_parameter< arma::vec const& >::type Event(EventSEXP);
    Rcpp::traits::input_parameter< arma::mat const& >::type DesignMat(DesignMatSEXP);
    Rcpp::traits::input_parameter< String const& >::type PriorCV(PriorCVSEXP);
    Rcpp::traits::input_parameter< double const& >::type HypTheta(HypThetaSEXP);
    Rcpp::traits::input_parameter< double const& >::type Hyp1Gam(Hyp1GamSEXP);
    Rcpp::traits::input_parameter< double const& >::type Hyp2Gam(Hyp2GamSEXP);
    Rcpp::traits::input_parameter< String const& >::type Mixing(MixingSEXP);
    Rcpp::traits::input_parameter< String const& >::type BaseModel(BaseModelSEXP);
    Rcpp::traits::input_parameter< int >::type thin(thinSEXP);
    Rcpp::traits::input_parameter< int >::type lambdaPeriod(lambdaPeriodSEXP);
    Rcpp::traits::input_parameter< double >::type ar(arSEXP);
    rcpp_result_gen = Rcpp::wrap(HiddenRMWreg_BFoutlier(Chain, RefLambda, Time, Event, DesignMat, PriorCV, HypTheta, Hyp1Gam, Hyp2Gam, Mixing, BaseModel, thin, lambdaPeriod, ar));
    return rcpp_result_gen;
END_RCPP
}

static const R_CallMethodDef CallEntries[] = {
    {"_RMWreg_HiddenLogPriorTheta", (DL_FUNC) &_RMWreg_HiddenLogPriorTheta, 5},
    {"_RMWreg_HiddenRMWreg_MCMC", (DL_FUNC) &_RMWreg_HiddenRMWreg_MCMC, 28},
    {"_RMWreg_HiddenLogLik", (DL_FUNC) &_RMWreg_HiddenLogLik, 8},
    {"_RMWreg_HiddenRMWreg_DIC", (DL_FUNC) &_RMWreg_HiddenRMWreg_DIC, 6},
    {"_RMWreg_HiddenRMWreg_CaseDeletion", (DL_FUNC) &_RMWreg_HiddenRMWreg_CaseDeletion, 6},
    {"_RMWreg_HiddenRMWreg_BFoutlierObs", (DL_FUNC) &_RMWreg_HiddenRMWreg_BFoutlierObs, 15},
    {"_RMWreg_HiddenRMWreg_BFoutlier", (DL_FUNC) &_RMWreg_HiddenRMWreg_BFoutlier, 14},
    {NULL, NULL, 0}
};

RcppExport void R_init_RMWreg(DllInfo *dll) {
    R_registerRoutines(dll, NULL, CallEntries, NULL, NULL);
    R_useDynamicSymbols(dll, FALSE);
}
